clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -3068,11 +3107,217 @@ void MallocChecker::checkDeadSymbols(SymbolReaper &SymReaper,
+  }
+
+  return false;
+}
+
+static void collectDirectSmartOwningPtrFieldRegions(
+    const MemRegion *Base, QualType RecQT, CheckerContext &C,
+    SmallVectorImpl<const MemRegion *> &Out) {
+  if (!Base)
+    return;
+  const auto *CRD = RecQT->getAsCXXRecordDecl();
+  if (!CRD)
+    return;
+
+  for (const FieldDecl *FD : CRD->fields()) {
+    if (!isSmartOwningPtrType(FD->getType()))
+      continue;
+    SVal L = C.getState()->getLValue(FD, loc::MemRegionVal(Base));
+    if (const MemRegion *FR = L.getAsRegion())
+      Out.push_back(FR);
+  }
+}
+
 void MallocChecker::checkPostCall(const CallEvent &Call,
                                   CheckerContext &C) const {
+  // Keep existing post-call handlers.
   if (const auto *PostFN = PostFnMap.lookup(Call)) {
     (*PostFN)(this, C.getState(), Call, C);
-    return;
+  }
+
+  SmallVector<const MemRegion *, 8> SmartPtrFieldRoots;
+
+  for (unsigned I = 0, E = Call.getNumArgs(); I != E; ++I) {
+    const Expr *AE = Call.getArgExpr(I);
+    if (!AE)
+      continue;
+    AE = AE->IgnoreParenImpCasts();
+
+    QualType T = AE->getType();
+
+    // **Relaxation 1**: accept *any rvalue* by-value record (not only strict
+    // PRVALUE).
+    if (AE->isGLValue())
+      continue;
+
+    // By-value record only (no refs).
+    if (!T->isRecordType() || T->isReferenceType())
+      continue;
+
+    // **Relaxation 2**: accept common temp/construct forms but don't overfit.
+    const bool LooksLikeTemp =
+        isa<CXXTemporaryObjectExpr>(AE) || isa<MaterializeTemporaryExpr>(AE) ||
+        isa<CXXConstructExpr>(AE) || isa<InitListExpr>(AE) ||
+        isa<ImplicitCastExpr>(AE) ||   // handle common rvalue materializations
+        isa<CXXBindTemporaryExpr>(AE); // handle CXXBindTemporaryExpr
+    if (!LooksLikeTemp)
+      continue;
+
+    // Require at least one direct smart owning pointer field by type.
+    const auto *CRD = T->getAsCXXRecordDecl();
+    if (!CRD)
+      continue;
+    bool HasSmartPtrField = false;
+    for (const FieldDecl *FD : CRD->fields()) {
+      if (isSmartOwningPtrType(FD->getType())) {
+        HasSmartPtrField = true;
+        break;
+      }
+    }
+    if (!HasSmartPtrField)
+      continue;
+
+    // Find a region for the argument.
+    SVal VCall = Call.getArgSVal(I);
+    SVal VExpr = C.getSVal(AE);
+    const MemRegion *RCall = VCall.getAsRegion();
+    const MemRegion *RExpr = VExpr.getAsRegion();
+
+    const MemRegion *Base = RCall ? RCall : RExpr;
+    if (!Base) {
+      // Fallback: if we have a by-value record with unique_ptr fields but no
+      // region, mark all allocated symbols as escaped
+      ProgramStateRef State = C.getState();
+      RegionStateTy RS = State->get<RegionState>();
+      ProgramStateRef NewState = State;
+      for (auto [Sym, RefSt] : RS) {
+        if (RefSt.isAllocated() || RefSt.isAllocatedOfSizeZero()) {
+          NewState =
+              NewState->set<RegionState>(Sym, RefState::getEscaped(&RefSt));
+        }
+      }
```

Comment:
This does something very similar to `EscapeTrackedCallback`. Have you considered reusing some code?